packages/expo/android/src/main/java/expo/modules/ApplicationLifecycleDispatcher.kt

```diff
@@ -2,19 +2,22 @@ package expo.modules
 
 import android.app.Application
 import android.content.res.Configuration
+import expo.modules.core.ModulePriorities
 
 class ApplicationLifecycleDispatcher private constructor() {
   companion object {
     @JvmStatic
     fun onApplicationCreate(application: Application) {
       ExpoModulesPackage.packageList
+        .sortedByDescending { ModulePriorities.get(it::class.qualifiedName) }
         .flatMap { it.createApplicationLifecycleListeners(application) }
```

Comment:
We're duplicating the `ExpoModulesPackages`-related logic about extracting `ApplicationLifecycleListeners` int this and in `onConfigurationChanged` methods. I guess we can provide some helper function that would look like this:
```kt
fun getApplicationLifecycleListeners() {
  return ExpoModulesPackage.packageList
    .sortedByDescending { ModulePriorities.get(it::class.qualifiedName) }
    .flatMap { it.createApplicationLifecycleListeners(application) }
}
```
and then:
```suggestion
      getApplicationLifecycleListeners()
```